CONTRIBUTING.md

```diff
@@ -1,12 +1,379 @@
+        },
+```
+
+You can see that we have replaced `ElectricalMeasurement.cluster_id` from endpoint 1 in the `signature` dict with the name of our cluster that we created: `ElectricalMeasurementCluster` and on endpoint 2 we replaced `AnalogInput.cluster_id` with the implementation we created for that: `AnalogInputCluster`. This instructs Zigpy to use these `CustomCluster` derivatives instead of the normal cluster definitions for these clusters and this is why this part of the quirk is called `replacement`.
+
+Now lets put this all together. If you examine the device definition above you will see that we have defined our custom device, we defined the `signature` dict where we transcribed the `SimpleDescriptor` output we obtained when the device joined the network and we defined the `replacement` dict where we swapped the cluster ids for the culsters that we wanted to replace with the `CustomCluster` implementations that we created.
+
+# Contribution Guidelines
 
 - All code is formatted with black. The check format script that runs in CI will ensure that code meets this requirement and that it is correctly formatted with black. Instructions for installing black in many editors can be found here: <https://github.com/psf/black#editor-integration>
 
-- Capture the SimpleDescriptor log entries for each endpoint on the device. These can be found in the HA logs after joining a device and they look like this: `<SimpleDescriptor endpoint=1 profile=260 device_type=1026 device_version=0 input_clusters=[0, 1, 3, 32, 1026, 1280, 2821] output_clusters=[25]>`. This information can also be obtained from the zigbee.db if you want to take the time to query the tables and reconstitute the log entry. I find it easier to just remove and rejoin the device. ZHA entity ids are stable for the most part so it _shouldn't_ disrupt anything you have configured. These need to match what the device reports EXACTLY or zigpy will not match them when a device joins and the handler will not be used for the device.
+- Capture the SimpleDescriptor log entries for each endpoint on the device. These can be found in the HA logs after joining a device and they look like this: `<SimpleDescriptor endpoint=1 profile=260 device_type=1026 device_version=0 input_clusters=[0, 1, 3, 32, 1026, 1280, 2821] output_clusters=[25]>`. This information can also be obtained from the zigbee.db if you want to take the time to query the tables and reconstitute the log entry. I find it easier to just remove and rejoin the device. ZHA entity ids are stable for the most part so it _shouldn't_ disrupt anything you have configured. These need to match what the device reports EXACTLY or zigpy will not match them when a device joins and the handler will not be used for the device. You can also obtain this information from the device screen in HA for the device. The `Zigbee Device Signature` button will launch a dialog that contains all of the information necessary to create quirks.
```

Comment:
Log settings to get these might be helpful to add here, ie how to see the `SimpleDescriptor`.  Note, I IIRC I didn't see the `SimpleDescriptor` string but maybe I had the wrong log level or because I was using a cc2531?

The docker/local install methodology of testing might also be helpful to add to this document.

Reply:
my descriptor had a slightly different format:
```

#   <Optional endpoint=1 profile=260 device_type=1026 device_version=1
#       input_clusters=[0, 3, 1280, 65535, 1]
#       output_clusters=[0, 4, 3, 5, 1]
#       >
#  {
#   "node_descriptor": "<NodeDescriptor
#       byte1=2
#       byte2=64
#       mac_capability_flags=128
#       manufacturer_code=4151
#       maximum_buffer_size=127
#       maximum_incoming_transfer_size=100
#       server_mask=0
#       maximum_outgoing_transfer_size=100
#       descriptor_capability_field=0
#       >",
#   "endpoints": {
#     "1": {
#       "profile_id": 260,
#       "device_type": "0x0402",
#       "in_clusters": [
#         "0x0000",
#         "0x0001",
#         "0x0003",
#         "0x0500",
#         "0xffff"
#       ],
#       "out_clusters": [
#         "0x0000",
#         "0x0001",
#         "0x0003",
#         "0x0004",
#         "0x0005"
#       ]
#     }
#   },
#   "manufacturer": "ORVIBO",
#   "model": "895a2d80097f4ae2b2d40500d5e03dcc",
#   "class": "zigpy.device.Device"
# }
```

Reply:
and a reminder to turn off the logs too!

Reply:
Simple descriptor is logged by the `zigpy` but only if the device is not present in the database -- very new join. On rejoins the  full discovery is skipped, so you might not see it. And it also could be burried by all other logging noise. 
In zigpy logs, you should see `...new device joined...` and after that be on lookout for SimpleDescriptor messages